lib/utils.ts

```diff
@@ -2,24 +2,51 @@ import fetch from 'node-fetch';
 import crypto from 'crypto';
 import fs from 'fs-extra';
 import httpsProxyAgent from 'https-proxy-agent';
+import type { HttpsProxyAgentOptions } from 'https-proxy-agent';
 import path from 'path';
 import { spawnSync, SpawnSyncOptions } from 'child_process';
 import stream from 'stream';
+import url from 'url';
 
 import { log, wasReported } from './log';
 
-export async function downloadUrl(url: string, file: string): Promise<void> {
+function getProxyHost() {
+  return process.env.HTTPS_PROXY ??
+  process.env.https_proxy ??
+  process.env.HTTP_PROXY ??
+  process.env.http_proxy;
+}
+
+function getProxyCert() {
+  return process.env.HTTPS_PROXY_CERT ?? process.env.https_proxy_cert;
+}
+
+function getProxyKey() {
+  return process.env.HTTPS_PROXY_KEY ?? process.env.https_proxy_key;
+}
+
+export async function downloadUrl(urlToDownload: string, file: string): Promise<void> {
   log.enableProgress(path.basename(file));
   log.showProgress(0);
 
-  const proxy =
-    process.env.HTTPS_PROXY ??
-    process.env.https_proxy ??
-    process.env.HTTP_PROXY ??
-    process.env.http_proxy;
+  const proxyHost = getProxyHost();
+
+  let proxy: HttpsProxyAgentOptions | void;
+  
+    if (proxyHost) {
+      proxy = url.parse(proxyHost);
```

Comment:
`url.parse` is deprecated, use `new URL(proxyHost)` instead

Reply:
I can do that if you'd like. I was a bit worried about changing existing behavior. I see that htpss-proxy-agent still is using url.parse: https://github.com/TooTallNate/node-https-proxy-agent/blob/master/src/agent.ts#L34

Reply:
hm, I would not expect there to be a conflict, and I'd rather use non-deprecated functions. `url.parse` returns a new URL object anyway, so it should be a drop in replacement, and more secure from things like host spoofing ([according to Node.js at least](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost))

Reply:
@vicapow and @robertsLando I'd still like to see this resolved, unless @robertsLando has strong opinions otherwise